pygmt/tests/test_grdview.py

```diff
@@ -69,184 +67,167 @@ def test_grdview_wrong_kind_of_grid(xrgrid):
         fig.grdview(grid=dataset)
 
 
-@check_figures_equal()
-def test_grdview_with_perspective(gridfile, xrgrid):
+@pytest.mark.mpl_image_compare
+def test_grdview_with_perspective(gridfile):
     """
     Run grdview by passing in a grid and setting a perspective viewpoint with
     an azimuth from the SouthEast and an elevation angle 15 degrees from the
     z-plane.
     """
-    fig_ref, fig_test = Figure(), Figure()
-    fig_ref.grdview(grid=gridfile, perspective=[135, 15])
-    fig_test.grdview(grid=xrgrid, perspective=[135, 15])
-    return fig_ref, fig_test
+    fig = Figure()
+    fig.grdview(grid=gridfile, perspective=[135, 15], frame=True)
```

Comment:
> Not sure if I am missing something - can you explain why `projection` is not set in the grdview tests?

Good point @meghanrjones. The projection (-J) parameter is listed as a required argument in https://docs.generic-mapping-tools.org/6.2.0rc2/grdview.html, so we probably should add it to the test? Setting `projection="M6c"` here does help resolve the issue at https://github.com/GenericMappingTools/pygmt/pull/1154/files#r639259330, but I'm still wondering if there's an issue with the GMT C API's auto projection setting.

Reply:
There shouldn't be an auto projection setting unless it was set previously in a modern mode session 😰 I get an appropriate error when I run an equivalent gmt command, so it is odd that it passes without complaint for PyGMT. I can look into this unless you want to.
```
$gmt grdcut @earth_relief_01d_g -R-116/-109/-47/-44 -Gtmp.nc
$gmt grdview tmp.nc -p135/15
grdview [WARNING]: The -p option works best in consort with -J (and -R or a grid)
grdview [ERROR]: Must specify a map projection with the -J option
```

Reply:
I think the default projection is Q15c+ for geographic data and X15c for cartesian data.

Reply:
Ah, thanks for the clarification, feel free to look into it (I got a lunch seminar to go to soon). I get the same error running `grdview` on the command line, so it must be something with PyGMT's global session that set a projection before this test ran.

Reply:
> I think the default projection is Q15c+ for geographic data and X15c for cartesian data.

Is this a GMT default or a PyGMT default?

Reply:
Some testing shows that it's a modern mode GMT default. Always more to learn it seems.

Reply:
Ok, I've added `projection="Q15c+"` to the `grdview` test at 2962bad64bf7cd2eed338a296157a7dfd67c67a8 so it should be ok now. The auto/default projection issue can be discussed another time :slightly_smiling_face: